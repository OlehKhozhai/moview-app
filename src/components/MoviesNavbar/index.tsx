import React from 'react';
import cn from 'classnames';

import MoviesNavbarTabs from 'components/MoviesNavbarTabs';
import MoviesNavbarSortBy from 'components/MoviesNavbarSortBy';
import styles from './styles.module.scss';

type MoviesNavbarProps = {
  className?: string;
};

const MoviesNavbar: React.FC<MoviesNavbarProps> = ({ className }) => {
  return (
    <nav className={cn(styles.root, className)}>
      <MoviesNavbarTabs />
      <MoviesNavbarSortBy />
    </nav>
  );
};

export default React.memo(MoviesNavbar);
