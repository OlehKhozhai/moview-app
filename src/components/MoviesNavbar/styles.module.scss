@import 'assets/styles/mixins/colors';

.root {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: $primary-bg;
  color: $primary-color;
  padding: 0 60px;
  border-bottom: 2px solid $secondary-border;
}
