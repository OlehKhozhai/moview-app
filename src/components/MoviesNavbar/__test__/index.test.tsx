import React from 'react';
import { create } from 'react-test-renderer';

import MoviesNavbar from '../index';

jest.mock('components/MoviesNavbarTabs', () => ({
  __esModule: true,
  default: () => '---MoviesNavbarTabs---',
}));

jest.mock('components/MoviesNavbarSortBy', () => ({
  __esModule: true,
  default: () => '---MoviesNavbarSortBy---',
}));

describe('MoviesNavbar component', () => {
  test('should render MoviesNavbar component by default', () => {
    const component = create(<MoviesNavbar searchParams="" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
