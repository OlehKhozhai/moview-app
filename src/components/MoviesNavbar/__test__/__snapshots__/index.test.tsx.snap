// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesNavbar component should render MoviesNavbar component by default 1`] = `
<nav
  className="root"
>
  ---MoviesNavbarTabs---
  ---MoviesNavbarSortBy---
</nav>
`;
